--- a/cmake/external/abseil-cpp.cmake
+++ b/cmake/external/abseil-cpp.cmake
@@ -22,6 +22,7 @@ FetchContent_Declare(
     URL ${DEP_URL_abseil_cpp}
     URL_HASH SHA1=${DEP_SHA1_abseil_cpp}
     PATCH_COMMAND ${ABSL_PATCH_COMMAND}
+    FIND_PACKAGE_ARGS REQUIRED CONFIG NAMES absl
 )
 
 onnxruntime_fetchcontent_makeavailable(abseil_cpp)
--- a/cmake/external/eigen.cmake
+++ b/cmake/external/eigen.cmake
@@ -14,8 +14,11 @@ else ()
         FetchContent_Declare(
         eigen
         URL https://gitlab.com/libeigen/eigen/-/archive/d10b27fe37736d2944630ecd7557cefa95cf87c9/eigen-d10b27fe37736d2944630ecd7557cefa95cf87c9.zip
+        FIND_PACKAGE_ARGS REQUIRED CONFIG NAMES Eigen3
 		)
     endif()
-    FetchContent_Populate(eigen)
-    set(eigen_INCLUDE_DIRS  "${eigen_SOURCE_DIR}")
+    # FetchContent_Populate(eigen)
+    # set(eigen_INCLUDE_DIRS  "${eigen_SOURCE_DIR}")
+    onnxruntime_fetchcontent_makeavailable(eigen)
+    get_target_property(eigen_INCLUDE_DIRS Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
 endif()